ouroboros-consensus/src/Ouroboros/Consensus/Storage/ImmutableDB/Impl.hs

```diff
@@ -800,7 +815,8 @@ startNewChunk tempRegistry hasFS index chunkInfo = do
       Index.appendOffsets index currentPrimaryHandle backfillOffsets
       `finally` cleanUp hasFS st
```

Comment:
Unrelated to this PR: we always call `cleanUp` to close the open handles of the current epoch. However, `cleanUp` also calls `Index.close`, which stops the background thread that expires past epochs from the cache! So we are leaking memory!

Reply:
Make sure that we have a test that catches this bug. Create a ticket for this if necessary.

Reply:
#1794.